src/components/Tabs/Tabs.tsx

```diff
@@ -12,15 +13,45 @@ interface ITabsProps {
 }
 
 export const Tabs = ({ children }: ITabsProps) => {
+  const [showScrollButton, setShowScrollButton] = useState(false);
+  const tabContent = document.getElementById('tab-content');
+
+  const onResize = useCallback(() => {
+    tabContent && setShowScrollButton(tabContent.scrollWidth > window.innerWidth);
+  }, [tabContent]);
```

Comment:
`tabContent` is assigned on each render, so onResize is recreated on each render.
Then useEffect handler below is rerun on each render as well.
Isn't `resize` event listener enough?

Reply:
the reason for calling `onResize` here is to initialize the scroll button when first loaded. Narrow-screen user will not be able to see the button without calling it at the very beginning(because they actually never "resize" the window.